### DISCUSSION TITLE
Refresh the page and disconnect using disconnect() cause page crash with error "Error: No matching key", guess its because the session key is lost after refreshing, how can i handle this?

### DISCUSSION THREAD

**Discussion by jingxuan98** [WalletConnect/walletconnect-monorepo#2857] [Developer Support] [answered]
Refresh the page and disconnect using disconnect() cause page crash with error 

"Error: No matching key. history: 1688012026990530"

![Screenshot 2023-06-29 at 12 15 05 PM](https://github.com/WalletConnect/walletconnect-monorepo/assets/43902263/c85d18fc-dba4-4632-823e-00931a177be1)

Guess its because the session key is lost after refreshing and disconenct() causing the issue, how can i handle this? 

using 
"@web3modal/standalone": "^2.4.3",
"@walletconnect/universal-provider": "^2.8.4"
"@walletconnect/types": "^2.8.4"

**Comment by glitch-txs** [comment]
@web3modal/standalone was replaced by @walletconnect/modal

https://docs.walletconnect.com/2.0/web/providers/ethereum

**Reply by sachin-kiwi** [reply]
@glitch-txs   your mentioned link is giving page not found

**Comment by siosio34** [comment]
I also have the same issue connecting walletconnect through wagmi and ethereum-provider.

**Reply by crypblizz8** [reply]
Hey there, we are working on better stack tracing on this error but if you clear your localStorage does that help resolve the issue?

**Comment by glitch-txs** [comment, marked as answer]
Aswered [here](https://github.com/orgs/WalletConnect/discussions/3291). Seems to be an issue on the Wallet side.
